### PR TITLE
Note support for Python 3.9

### PR DESCRIPTION
**Pull request by callahad**
As expected, all tests pass locally without modification.

Signed-off-by: Dan Callahan <danc@element.io>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by callahad** [comment]
We should add 3.9 to our buildkite pipeline over in https://github.com/matrix-org/pipelines/blob/master/synapse/pipeline.yml, but I'm not sure what mix of other bits we should add with it.

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = packaging, py35, py36, py37, py38, check_codestyle, check_isort
+envlist = packaging, py35, py36, py37, py38, py39, check_codestyle, check_isort
```
Comment (anoadragon453) [inline tox.ini:2]: Are these environments actually used if they're not repeated below :thinking:

**Review by anoadragon453 — APPROVED** [review]
I looked through the closed PRs list for what we did for Python 3.8 and 3.7. I think the only thing left is updating the Python version in the docker container (the default is still 3.7!): https://github.com/matrix-org/synapse/blob/2e380f0f1802e5310f5a19b5cac314ce742aaecd/docker/Dockerfile#L14

Would you be able to provide a PR for the pipeline.yml file as well? I'm not entirely sure which SyTest configuration to use Python 3.9 in, but we can discuss it with everyone in #synapse-dev :)

File: tox.ini (hunk 1 above)
Comment (callahad) [reply tox.ini:2]: My understanding is that tox uses that envlist and the default `[testenv]` section when invoked without an explicit `-e` parameter on the commandline.

**Comment by callahad** [comment]
> I think the only thing left is updating the Python version in the docker container (the default is still 3.7!)

That feels like a slightly separate question — we're fine supporting Py 3.9, but do we actually want to *ship* it just yet? :)

> Would you be able to provide a PR for the pipeline.yml file as well?

Yep! I've manually sketched out our testing matrix and will create an issue over in the pipelines repo for discussion soon.

File: tox.ini (hunk 1 above)
Comment (anoadragon453) [reply tox.ini:2]: Ah indeed, TIL!

**Comment by anoadragon453** [comment]
> That feels like a slightly separate question — we're fine supporting Py 3.9, but do we actually want to ship it just yet? :)

Good point - we probably at least want to get our full CI using it for a bit before doing so.

I'm not sure what the timeline for that looks like, though we may want to consider at least bumping it to 3.8 in the meantime. An issue for discussion in this repo as well would be helpful I think.

> Yep! I've manually sketched out our testing matrix and will create an issue over in the pipelines repo for discussion soon.

Wonderful, thanks!

**Comment by anoadragon453** [comment]
Other that those orthogonal discussions then, this PR looks good to go :+1:

**Comment by callahad** [comment]
Question on updating Dockerfile to Python 3.9 is at #8674 

Question on where Python 3.9 should fit into our test matrix at matrix-org/pipelines#104

**Pull request merged**